apps/rpc/src/modules/company/company-router.ts

```diff
@@ -1,68 +1,101 @@
-      ctx.executeTransaction(async (handle) => ctx.companyService.findMany(handle, input))
-    ),
+export type FindCompanyByIdInput = inferProcedureInput<typeof findCompanyByIdProcedure>
+export type FindCompanyByIdOutput = inferProcedureOutput<typeof findCompanyByIdProcedure>
+const findCompanyByIdProcedure = procedure
+  .input(CompanySchema.shape.id)
+  .use(withDatabaseTransaction())
+  .query(async ({ input, ctx }) => ctx.companyService.findById(ctx.handle, input))
 
-  findById: procedure
-    .input(CompanySchema.shape.id)
-    .query(async ({ input, ctx }) =>
-      ctx.executeTransaction(async (handle) => ctx.companyService.findById(handle, input))
-    ),
+export type GetCompanyByIdInput = inferProcedureInput<typeof getCompanyByIdProcedure>
+export type GetCompanyByIdOutput = inferProcedureOutput<typeof getCompanyByIdProcedure>
+const getCompanyByIdProcedure = procedure
+  .input(CompanySchema.shape.id)
+  .use(withDatabaseTransaction())
+  .query(async ({ input, ctx }) => ctx.companyService.getById(ctx.handle, input))
 
-  getById: procedure
-    .input(CompanySchema.shape.id)
-    .query(async ({ input, ctx }) =>
-      ctx.executeTransaction(async (handle) => ctx.companyService.getById(handle, input))
-    ),
+export type FindCompanyBySlugInput = inferProcedureInput<typeof findCompanyBySlugProcedure>
+export type FindCompanyBySlugOutput = inferProcedureOutput<typeof findCompanyBySlugProcedure>
+const findCompanyBySlugProcedure = procedure
+  .input(CompanySchema.shape.slug)
+  .use(withDatabaseTransaction())
+  .query(async ({ input, ctx }) => ctx.companyService.findBySlug(ctx.handle, input))
 
-  findBySlug: procedure
-    .input(CompanySchema.shape.slug)
-    .query(async ({ input, ctx }) =>
-      ctx.executeTransaction(async (handle) => ctx.companyService.findBySlug(handle, input))
-    ),
+export type GetCompanyBySlugInput = inferProcedureInput<typeof getCompanyBySlugProcedure>
+export type GetCompanyBySlugOutput = inferProcedureOutput<typeof getCompanyBySlugProcedure>
+const getCompanyBySlugProcedure = procedure
+  .input(CompanySchema.shape.slug)
+  .use(withDatabaseTransaction())
+  .query(async ({ input, ctx }) => ctx.companyService.getBySlug(ctx.handle, input))
 
-  getBySlug: procedure
-    .input(CompanySchema.shape.slug)
-    .query(async ({ input, ctx }) =>
-      ctx.executeTransaction(async (handle) => ctx.companyService.getBySlug(handle, input))
-    ),
+export type CreateCompanyFileUploadInput = inferProcedureInput<typeof createCompanyFileUploadProcedure>
+export type CreateCompanyFileUploadOutput = inferProcedureOutput<typeof createCompanyFileUploadProcedure>
+const createCompanyFileUploadProcedure = procedure
+  .input(
+    z.object({
+      filename: z.string(),
+      contentType: z.string(),
+    })
+  )
+  .output(z.custom<PresignedPost>())
+  .use(withAuthentication())
+  .use(withAuthorization(isEditor()))
+  .use(withDatabaseTransaction())
+  .use(withAuditLogEntry())
```

Comment:
this doesnt do anything as the route does not really use the database at all, but its fine